source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1144 +25,3357 @@
+ * options and a pointer to a F_TCP_UDP_Handler_t.
+ *
+ * FREERTOS_SO_TCP_CONN_HANDLER  * Callback for (dis)connection events. *
+ * FREERTOS_SO_TCP_RECV_HANDLER  * Callback for receiving TCP data. *
+ * FREERTOS_SO_TCP_SENT_HANDLER  * Callback for sending TCP data. *
+ * FREERTOS_SO_UDP_RECV_HANDLER  * Callback for receiving UDP data. *
+ * FREERTOS_SO_UDP_SENT_HANDLER  * Callback for sending UDP data. *
+ *
+ * typedef struct xTCP_UDP_HANDLER
+ * {
+ *      FOnConnected_t pxOnTCPConnected; * FREERTOS_SO_TCP_CONN_HANDLER *
+ *      FOnTCPReceive_t pxOnTCPReceive;  * FREERTOS_SO_TCP_RECV_HANDLER *
+ *      FOnTCPSent_t pxOnTCPSent;        * FREERTOS_SO_TCP_SENT_HANDLER *
+ *      FOnUDPReceive_t pxOnUDPReceive;  * FREERTOS_SO_UDP_RECV_HANDLER *
+ *      FOnUDPSent_t pxOnUDPSent;        * FREERTOS_SO_UDP_SENT_HANDLER *
+ * } F_TCP_UDP_Handler_t
+ *
+ * Function Prototypes:
+ *
+ * typedef void (* FOnConnected_t )( Socket_t xSocket,
+ *                                   BaseType_t ulConnected )
+ *
+ * typedef BaseType_t (* FOnTCPReceive_t )( Socket_t xSocket,
+ *                                          void * pData,
+ *                                          size_t xLength )
+ *
+ * typedef void (* FOnTCPSent_t )( Socket_t xSocket,
+ *                                 size_t xLength )
+ *
+ * typedef BaseType_t (* FOnUDPReceive_t ) ( Socket_t xSocket,
+ *                                           void * pData,
+ *                                           size_t xLength,
+ *                                           const struct freertos_sockaddr * pxFrom,
+ *                                           const struct freertos_sockaddr * pxDest )
+ *
+ * typedef void (* FOnUDPSent_t )( Socket_t xSocket,
+ *                                      size_t xLength );
+ */
+
+#ifndef ipconfigUSE_CALLBACKS
+    #define ipconfigUSE_CALLBACKS    ipconfigDISABLE
+#endif
+
+#if ( ( ipconfigUSE_CALLBACKS != ipconfigDISABLE ) && ( ipconfigUSE_CALLBACKS != ipconfigENABLE ) )
+    #error Invalid ipconfigUSE_CALLBACKS configuration
+#endif
+
+/*---------------------------------------------------------------------------*/
+
+/*#if ( ipconfigIS_ENABLED( ipconfigUSE_CALLBACKS ) )*/
+
+/*-----------------------------------------------------------------------*/
+
+/*
+ * ipconfigIS_VALID_PROG_ADDRESS
+ *
+ * https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_IP_Configuration.html#ipconfigIS_VALID_PROG_ADDRESS
+ *
+ * Type: Macro Function
+ * Value: BaseType_t ( pdTRUE | pdFALSE )
+ *
+ * Verifies that a given address refers to valid (instruction) memory.
+ *
+ * Used to check if application defined hooks are valid.
+ *
+ * Example:
+ * if( ipconfigIS_VALID_PROG_ADDRESS( pxSocket->u.xTCP.pxHandleSent ) )
+ * {
+ *     pxSocket->u.xTCP.pxHandleSent( pxSocket, ulCount );
+ * }
+ */
+
+#ifndef ipconfigIS_VALID_PROG_ADDRESS
+    #define ipconfigIS_VALID_PROG_ADDRESS( pxAddress )    ( pxAddress != 0 )
```

Comment:
```suggestion
    #define ipconfigIS_VALID_PROG_ADDRESS( pxAddress )    ( ( pxAddress ) != NULL )
```
I'd suggest to wrap the variable in macro and compare the pointer with NULL instead of 0.